addons/stock/models/stock_scrap.py

```diff
@@ -70,6 +70,10 @@ def _compute_location_id(self):
         for scrap in self:
             if scrap.picking_id:
                 scrap.location_id = scrap.picking_id.location_dest_id if scrap.picking_id.state == 'done' else scrap.picking_id.location_id
+            elif not locations_per_company:
+                raise UserError(_(
+                    "Warehouse for the company '%s' is deleted or archived, please create a new one or unarchive it.", scrap.company_id.name
+                ))
```

Comment:
No need to raise an UseErro; use `active_test=False` for read group of  `stock.warehouse` like
```
groups = self.env['stock.warehouse'].with_context(active_test=False)._read_group(
            [('company_id', 'in', self.company_id.ids)], ['company_id'], ['lot_stock_id:array_agg'])
```

Reply:
bss yahi karte rehenge